crates/redstone/src/helpers/slice.rs

```diff
@@ -0,0 +1,101 @@
+/// Performs lookup for repeated value in the list.
+///
+/// # Arguments
+///
+/// * `list` - A `&[T]` list where `T` implements `PartialEq`, `Eq`, `Copy` and `Ord` traits.
+///
+/// # Returns
+///
+/// Returns a `Option<T>` which contains first repeated element found in the list or if there is no repeated
+/// element then None otherwise.
+pub fn has_duplicates<T>(list: &[T]) -> Option<T>
```

Comment:
Since we return an option here but use it like an error on the call site and it imho a little bit akward. Wdyt about such a change:
`fn check_no_duplicates<T>(list: &[T]) -> Result<(), T> // Err(first_duplication)`

Reply:
Let me think about it. Sounds like good idea!

Reply:
Because function `check_no_duplicates` doesn't know about the matter of type and type related error I decided to keep it simple with Option<T>. 
But maybe I am not entirely understanding your suggestion @kostekIV .  Please free to elaborate a bit more or better give some example. I am open to apply the change.

Reply:
U Right mate. On my way.